UI/SettingsWindow.cs

```diff
@@ -168,9 +170,41 @@ bool validateImgurClientId()
         return true;
       }
 
-      if (checkBoxArtworkUpload.Checked && textBoxImgurClientId.Text.Length > 0 && !validateImgurClientId())
+      bool validateS3()
       {
-        return false;
+        if (string.IsNullOrWhiteSpace(_settings.S3AccessKeyId) ||
+              string.IsNullOrWhiteSpace(_settings.S3SecretAccessKey) ||
+              string.IsNullOrWhiteSpace(_settings.S3BucketName))
```

Comment:
**issue:** This should also be checking `_settings.S3Endpoint`. Looks like I forgot to add that in the original changes 😅 

```suggestion
        if (string.IsNullOrWhiteSpace(_settings.S3AccessKeyId) ||
              string.IsNullOrWhiteSpace(_settings.S3SecretAccessKey) ||
              string.IsNullOrWhiteSpace(_settings.S3BucketName) ||
              string.IsNullOrWhiteSpace(_settings.S3Endpoint))
```